Sporadic test failure in GarbageCollectorTests.testIOException

**Describe the bug**
Found this test failing sporadically: https://github.com/pravega/pravega/runs/2621520353 
```
io.pravega.segmentstore.storage.chunklayer.GarbageCollectorTests > testIOException FAILED
    java.lang.AssertionError: expected:<1> but was:<0>
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:645)
        at org.junit.Assert.assertEquals(Assert.java:631)
        at io.pravega.segmentstore.storage.chunklayer.GarbageCollectorTests.assertQueueEquals(GarbageCollectorTests.java:996)
        at io.pravega.segmentstore.storage.chunklayer.GarbageCollectorTests.testIOException(GarbageCollectorTests.java:589)
```

**To Reproduce**
Running `gradlew build` may reproduce the problem sporadically.

**Screenshots**
n/a

**Additional information**
We need to check that this is just a flaky test and not a real bug.

Another instance of this flaky test: https://github.com/pravega/pravega/runs/4168738827?check_suite_focus=true
```
io.pravega.segmentstore.storage.chunklayer.GarbageCollectorTests > testIOException FAILED
    java.lang.AssertionError: expected:<1> but was:<0>
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:645)
        at org.junit.Assert.assertEquals(Assert.java:631)
        at io.pravega.segmentstore.storage.chunklayer.GarbageCollectorTests.assertQueueEquals(GarbageCollectorTests.java:1325)
        at io.pravega.segmentstore.storage.chunklayer.GarbageCollectorTests.testIOException(GarbageCollectorTests.java:644)
```

--- a/segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/chunklayer/GarbageCollectorTests.java
+++ b/segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/chunklayer/GarbageCollectorTests.java
@@ -608,7 +608,7 @@ public class GarbageCollectorTests extends ThreadPooledTestSuite {
         Function<Duration, CompletableFuture<Void>> noDelay = d -> CompletableFuture.completedFuture(null);
         val testTaskQueue = new InMemoryTaskQueueManager();
 
-        chunkStorage.setReadOnly(chunkStorage.openWrite("deletedChunk").get(), true);
+        chunkStorage.setReadOnly(chunkStorage.openWrite("deletedChunk").get(), true).join();
 
         @Cleanup
         GarbageCollector garbageCollector = new GarbageCollector(containerId,

